data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/SingleQueryParameterUtils.java

```diff
@@ -74,12 +75,23 @@ public static UInt64 getParameterValueAsUInt64(
     }
   }
 
+  public static UInt64 getParameterValueAsEpoch(
+      final Map<String, List<String>> parameterMap, final String key)
+      throws IllegalArgumentException {
+    final UInt64 value = getParameterValueAsUInt64(parameterMap, key);
+    // Restrict valid epoch values to ones that can be converted to slot without overflowing
+    if (value.isGreaterThan(UInt64.MAX_VALUE.dividedBy(Constants.SLOTS_PER_EPOCH))) {
```

Comment:
can we just store UInt64.MAX_VALUE.dividedBy(Constants.SLOTS_PER_EPOCH)) as a const? it's going to potentially happen a lot...

Reply:
It will happen at most once per RPC request and it equates to dividing a long, so we don't need to worry about the performance.

I tend to avoid putting `Constants` values into constants because ironically, they aren't actually constant - they're defined by the network definition. `Constants` get set very early on in the startup sequence so generally they have the right values before static initialisers run but there's nothing that guarantees that and it always makes me worry that they'll get initialised in the wrong order and cause trouble.